### ISSUE TITLE
Sprint: implement Grok-1 route-preservation and block-drift metrics

### ISSUE THREAD

**Issue by rmems** [rmems/xai-dissect#29]
## Context

For Grok-1 MoE quantization, generic tensor reconstruction error is not enough. The main failure mode is route drift: quantization may preserve weights numerically while changing expert selection behavior.

This issue defines the validation metrics needed before any full-checkpoint quantization run. Vultr credits expire on **2026-05-28**, so metrics should be implemented before expensive runs.

## Goal

Add MoE-specific validation metrics for pilot quantization artifacts.

### Weight reconstruction

- [ ] MSE
- [ ] cosine similarity
- [ ] max absolute error
- [ ] per-channel scale/error summary where available

### Block behavior

- [ ] block output cosine similarity
- [ ] block output RMSE
- [ ] residual stream drift summary

### Router behavior

- [ ] top-1 expert match rate
- [ ] top-2 expert set match rate
- [ ] expert-load distribution delta
- [ ] KL or JS divergence over expert-load distribution
- [ ] router-logit rank correlation where logits are available

### Model/logit behavior, if available

- [ ] logit KL against baseline
- [ ] perplexity delta on calibration data
- [ ] short generation sanity summary

## Suggested first-pass thresholds

```text
router top-1 agreement:      >= 99.0%
router top-2 set agreement:  >= 99.5%
block output cosine:         >= 0.995
perplexity delta:            report-only at first
logit KL:                    report-only at first
```

## Tasks

- [ ] Define metric structs/schema.
- [ ] Add CLI/report output for pilot comparisons.
- [ ] Add JSON output for machine processing.
- [ ] Add Markdown summary output for human review.
- [ ] Make route-preservation metrics first-class in the report.
- [ ] Add tests for metric computation on small fixtures.

## Acceptance criteria

- [ ] A pilot run emits a single route-preservation report.
- [ ] The report clearly marks pass/fail/unknown for each metric.
- [ ] Router metrics are shown above generic tensor-error metrics.
- [ ] The report can be attached to a PR or GitHub issue as sprint evidence.

**Comment by rmems** [comment]
Linked Linear tracker: MET-93 in the `xai-dissect + grok-ozempic Rust sprint` project.

This is the planning-side counterpart for route-preservation and block-drift metrics.

**Comment by rmems** [comment]
Implemented and pushed on branch `feat/grok1-pr-issues-25-31`. Added first-class route-preservation report artifacts: schema, CLI command, Markdown/JSON writers, output-tree bundle support, thresholds/metric surface, docs, and snapshot tests. Status values remain planning-surface driven until downstream pilot evidence populates observed metrics. Tests pass locally via `cargo test --quiet`.
